templates/hostname-RedHat.erb

```diff
@@ -10,4 +10,6 @@ NOZEROCONF="<%= @nozeroconf %>"
 NETWORKING_IPV6="<%= @ipv6enable %>"
 IPV6INIT="<%= @ipv6enable %>"
 <% end -%>
-HOSTNAME="<%= @manage_hostname.split('.').first %>"
+<% if @hostname -%>
+HOSTNAME="<%= @hostname %>"
+<% end -%>
```

Comment:
So, here we have an original sin that I'm hesitant to change, what I would do is to keep by default this (wrong) behaviour and add the possibility to make the right thing with a dedicated parameter.